Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -849,6 +849,148 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Resolve Conflict SG", "[.SyncServe
+                    it.key().asstring(),
+                    it.value()
+                    );
+            c4Params.optionsDictFleece = updated.data();
+        }
+    };
+
+    // Setup onDocsEnded:
+    _enableDocProgressNotifications = true;
+    _onDocsEnded = [](C4Replicator* repl,
+                      bool pushing,
+                      size_t numDocs,
+                      const C4DocumentEnded* docs[],
+                      void* context) {
+        for (size_t i = 0; i < numDocs; ++i) {
+            auto doc = docs[i];
+            if ((doc->flags & kRevPurged) == kRevPurged) {
+                ((ReplicatorAPITest*)context)->_docsEnded++;
+            }
+        }
+    };
+
+    auto collRoses = collections[0];
+
+    auto collRosesPath = repl::Options::collectionSpecToPath(collectionSpecs[0]);
+
+    // Put doc in remote DB, in channels a and b
+    sendRemoteRequest("PUT", collRosesPath, docIDstr, &status, &error,
+                      addChannelToJSON("{}", "channels"_sl, { channelIDa, channelIDb }));
+    REQUIRE(status == HTTPStatus::Created);
+
+    // Pull doc into CBL:
+    C4Log("-------- Pulling");
+    replicate(paramsSetter);
+
+    // Verify:
+    c4::ref<C4Document> doc1 = c4coll_getDoc(collRoses, slice(docIDstr), true, kDocGetAll, nullptr);
+    REQUIRE(doc1);
+    CHECK(slice(doc1->revID).hasPrefix("1-"_sl));
+    CHECK(_docsEnded == 0);
+    CHECK(_counter == 0);
+
+    // Revoked access to channel 'a':
+    sendRemoteRequest("PUT", "_user/purgeRevoke", &status, &error,
+                      addChannelToJSON("{}", "admin_channels"_sl, { channelIDb }), true);
+    REQUIRE(status == HTTPStatus::OK);
+
+    // Check if update to doc1 is still pullable:
+    auto oRevID = slice(doc1->revID).asString();
+    sendRemoteRequest("PUT", collRosesPath, docIDstr, &status, &error,
+                       addChannelToJSON(R"({"_rev":")" + oRevID + "\"}", "channels"_sl, { channelIDb }));
+
+    C4Log("-------- Pull update");
+    replicate(paramsSetter);
+
+    // Verify the update:
+    doc1 = c4coll_getDoc(collRoses, slice(docIDstr), true, kDocGetAll, nullptr);
+    REQUIRE(doc1);
+    CHECK(slice(doc1->revID).hasPrefix("2-"_sl));
+    CHECK(_docsEnded == 0);
+    CHECK(_counter == 0);
+
+    auto curRevID = slice(doc1->revID).asString();
+
+    // Revoke access to all channels:
+    sendRemoteRequest("PUT", "_user/purgeRevoke", &status, &error,
+                      addChannelToJSON("{}", "admin_channels"_sl, { }), true);
+    REQUIRE(status == HTTPStatus::OK);
+
+    C4Log("-------- Pull the revoked");
+    replicate(paramsSetter);
+
+    // Verify that doc1 is purged:
+    doc1 = c4coll_getDoc(collRoses, slice(docIDstr), true, kDocGetAll, nullptr);
+    REQUIRE(!doc1);
+    CHECK(_docsEnded == 1);
+    CHECK(_counter == 1);
+
+    // Purge remote doc so test will succeed multiple times without flushing bucket
+//    sendRemoteRequest("POST", "_purge", &status, &error, "{\"" + docIDstr + "\":[\"*\"]}", true);
```

Comment:
If this is not necessary for cleaning up, you can remove it.

Reply:
Ah, I tried to look through the code for any parts I'd left commented, thank you for spotting!